src/main/java/org/spongepowered/api/command/Command.java

```diff
@@ -0,0 +1,312 @@
+         *                 their respective {@link Command} children.
+         * @return This builder, for chaining
+         */
+        Builder addChildren(Map<? extends Iterable<String>, ? extends Command> children);
+
+        /**
+         * If this is set to true, then if the parent command (this) requires
+         * a {@link CommandSource} to have a permission
+         * (see {@link #permission(String)}), this permission is
+         * required for all children too. If this is set to false, then
+         * child commands <em>do not</em> require this permission.
+         *
+         * <p>This defaults to {@code true}.</p>
+         *
+         * @param required Whether this command's permission is required for
+         *                 child commands.
+         * @return This builder, for chaining
+         */
+        Builder requirePermissionForChildren(boolean required);
+
+        /**
+         * Determines what to do if a child command throws an exception.
+         *
+         * <p>Defaults to {@link ChildExceptionBehaviors#SUPPRESS}, which means
+         * that if a child command fails to execute, the error will be ignored
+         * and this base command will attempt to execute.</p>
+         *
+         * <p>See {@link ChildExceptionBehaviors} for other possible behaviors.
+         * </p>
+         *
+         * @param exceptionBehavior The {@link ChildExceptionBehavior} to adhere
+         *                          to.
+         * @return This builder, for chaining
+         */
+        Builder childExceptionBehavior(ChildExceptionBehavior exceptionBehavior);
+
+        /**
+         * Provides a simple description for this command, typically no more
+         * than one line.
+         *
+         * <p>Fuller descriptions should be provided through
+         * {@link #extendedDescription(Text)}</p>
+         *
+         * @param description The description to use, or {@code null} for no
+         *                    description
+         * @return This builder, for chaining
+         */
+        Builder description(@Nullable Text description);
+
+        /**
+         * Provides the logic of the command.
+         *
+         * <p>This is only optional if child commands are specified.</p>
+         *
+         * @param executor The {@link CommandExecutor} that will run the command
+         * @return This builder, for chaining
+         */
+        Builder executor(CommandExecutor executor);
+
+        /**
+         * Provides the description for this command.
+         *
+         * <p>A one line summary should be provided to
+         * {@link #description(Text)}</p>
+         *
+         * @param extendedDescription The description to use, or {@code null}
+         *                            for no description.
+         * @return This builder, for chaining
+         */
+        Builder extendedDescription(@Nullable Text extendedDescription);
```

Comment:
If we are breaking, can we change this to getHelp() , it's always annoyed me trying to recreate getHelp from the old extended descriptions.

Either that, or Command should have extendedDescription and leave it up to the help system to craft the help from it.